packages/components/src/custom-select-control-new/custom-select-control/hook.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import { useSelectState } from 'ariakit/select';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import * as styles from '../styles';
+import type { WordPressComponentProps } from '../../ui/context';
+import { useCx } from '../../utils/hooks/use-cx';
+import { SelectControlProps } from '../types';
+
+// TODO:
+// - should we use 'select' instead of `div` for props inheritance?
+// - should we allow polymorphism ?
```

Comment:
I wouldn't really expect `CustomSelectControlProps` to inherit from `HTMLSelectElement` at all since it's completely custom and doesn't ultimately render a `<select>`, no? That is, I'd expect the type to be `SelectControlProps` and not `WordPressComponentProps< SelectControlProps, 'select', false >`.

Reply:
> I wouldn't really expect `CustomSelectControlProps` to inherit from `HTMLSelectElement` at all since it's completely custom and doesn't ultimately render a `<select>`, no?

This in an interesting one. On one end, we should aim at `CustomSelectControlProps` to be semantically a replacement for the `select` HTML element — on the other end, it's not going to be easy to support all props/functionality of the native `select` HTML element. 

For example, `Ariakit` doesn't seem to support all of those props either:

https://github.com/ariakit/ariakit/blob/ariakit%402.0.0-next.36/packages/ariakit/src/select/select.tsx#L285

> That is, I'd expect props to have type SelectControlProps and not WordPressComponentProps< SelectControlProps, 'select', false >.

By default, components in the package should use the `WordPressComponentProps` type utility unless there's a good reason for them not to. At most, if we wanted to keep it as general as possible, we could type the props as `WordPressComponentProps< SelectControlProps, 'div', false >`